### PR TITLE
Send batches of docs to the input queue to reduce CPU load

### PR DESCRIPTION
**Pull request by stiegerb**
Adding an option (`--query_queue_batch_size`, default 5) to gather up a batch of documents before passing them to the input queue and bunching them up in bigger bunches to be uploaded. This reduces the CPU load of the query processes a bit. I'm still trying to find an optimal size, but 5 is already better than single documents.

### CODE REVIEW

**Comment by stiegerb** [comment]
@bbockelm It seems this also reduces the memory consumption, though I don't understand why.

File: spider_cms.py [hunk 1]
```diff
@@ -643,16 +651,16 @@ def run(self):
...
 
             if self.count_in%self.report_every == 0:
                 logging.debug("Processed %d docs" % self.count_in)
 
             # If buffer is full, send the docs and clear the buffer
-            if len(self.buffer) == self.bunch_size:
-                self.output_queue.put(self.buffer)
-                self.buffer = []
+            if len(self.buffer) >= self.bunch_size:
+                self.output_queue.put(self.buffer[:self.bunch_size])
```
Comment (bbockelm) [inline spider_cms.py:662]: Why not just put the whole buffer?

**Comment by bbockelm** [comment]
Not clear to me why we don't just send the whole bunch... but in the interest of time, going ahead with the merge.

File: spider_cms.py (hunk 1 above)
Comment (stiegerb) [reply spider_cms.py:662]: No strong reason, just that this way the bunches have always exactly the configured bunch size.

**Pull request merged**